src/app/shared/components/grid-list/grid-list.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, Input, OnInit } from "@angular/core";
+
+@Component({
+    selector: "app-grid-list",
+    templateUrl: "./grid-list.component.html",
+    styleUrls: ["./grid-list.component.scss"]
+})
+export class GridListComponent implements OnInit {
+    @Input() mobileColumnCount: number = 1;
+    @Input() itemColumnSpan: number = 1;
+
+    constructor() { }
+
+    ngOnInit(): void {
+        this.mobileColumnCount = this.clampColumnValue(this.mobileColumnCount);
+        this.itemColumnSpan = this.clampColumnValue(this.itemColumnSpan);
+    }
+
+    getColumnClass(): string {
+        return `mobile-columns-${ this.mobileColumnCount }`;
+    }
+
+    getItemColumnSpan(): string {
+        return `item-span-${ this.itemColumnSpan }`;
+    }
+
+    private clampColumnValue(columnValue: number): number {
```

Comment:
I like this method. It seems likely that we will use it some other place correct?

This is the kind of transformation that is a great use case for a pipe, pipes can also be used in component files by injecting them in the constructor and calling the transform method. I will explain in another ticket